### DISCUSSION TITLE
Still logging "return data is undefined" even null data is returned in useQuery ( Next.js)

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [TanStack/query#5893] [Q&A] [answered]
### Describe the bug

When fetching data with useQuery and Data have error . In console log it says undefined is returned . 
I display error message in ui but .
how to handle this undefined issue in Devtools ?   

![Screenshot (280)](https://github.com/TanStack/query/assets/71011043/7d637c67-2ff5-4134-85fa-2704d1316706)

### Version
Latest

**Comment by Arcades2** [comment]
Hello,

Can you show the definition of your `fetchAllUsers` function ?

**Comment by thanhtutzaw** [comment]
> `fetchAllUsers`
### Currently this fetching will get error because of quota limit of Firebase

 
![image](https://github.com/TanStack/query/assets/71011043/10f22fc6-2917-41b9-ab22-5da842aeb151)

**Comment by Arcades2** [comment, marked as answer]
What is the value of `uid` ? If it is a falsy value, you enter in your `if` statement and actually return `undefined`
